python/torch_mlir/eager_mode/torch_mlir_eager_backend.py

```diff
@@ -0,0 +1,69 @@
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+import abc
+from dataclasses import dataclass
+from typing import TypeVar, Tuple, Callable, List, Dict, Any
+
+import torch
+
+from torch_mlir._mlir_libs._mlir.ir import Module
+
+# TODO: This might need to be an ABC too, such as
+# to support finding the backend that created the tensor.
+DeviceTensor = TypeVar("DeviceTensor")
+
+
+@dataclass
+class TensorMetaData:
```

Comment:
Can you document this? I don't expect all backends will be able to know requires_grad/device/layout/strides all the time -- is there a way that we can factor that out of here? I think that size/dtype is the only thing we should need for now.
Bonus points if you just use meta tensors to compute this, and don't even need this callback at all (except possibly to assert that it matches the meta tensor computation).

Reply:
I don't know how much of the metadata determines decisions made by the pytorch dispatcher, so I'm erring on the side of caution, in addition to having in mind the [comment that I linked](https://github.com/albanD/subclass_zoo/blob/1566e038f03cd89ab3cc37e670a44e3c2bbc1897/trivial_tensors.py#L90-L92):

>external wrapper tensor is still required to have all of the metadata that the inner tensor has; this includes **stride and storage**!

Regarding

>Bonus points if you just use meta tensors to compute this

Meta tensor coverage is "spotty", [by PyTorch's own admission](https://github.com/albanD/subclass_zoo/blob/30baf07886c8c8948d465996548295bf5f438d89/python_meta_tensor.py#L17), so this isn't feasible (even if it did give you all of the metadata).

Reply:
just another note here: without `requires_grad` you're not going to get the autodiff graph being constructed, so that one is definitely a non-negotiable.